cf/cognito.yaml

```diff
@@ -15,12 +15,59 @@ Resources:
   FileUploadIdentityPool:
     Type: AWS::Cognito::IdentityPool
     Properties:
-      IdentityPoolName: "file-upload-identity-pool"
+      IdentityPoolName: !Sub "file-upload-identity-pool-${Environment}"
       AllowUnauthenticatedIdentities: false
 
+  UIRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      RoleName: !Sub "ui-auth-user-role-${Environment}"
+      Path: /
+      Policies:
+        - PolicyName: !Sub "can-upload-object-to-destination-bucket-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "s3:PutObject"
+                  - "s3:ListMultipartUploadParts"
+                  - "s3:AbortMultipartUpload"
+                  - "s3:ListBucketMultipartUploads"
+                Resource:
+                  - !Sub "arn:aws:s3:::biomage-source-${Environment}/*"
+
+        - PolicyName: !Sub "can-get-object-information-in-bucket-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "s3:GetObject"
+                  - "s3:GetObjectTagging"
+                Resource:
+                  - !Sub "arn:aws:s3:::biomage-source-${Environment}/*"
+
+        - PolicyName: !Sub "can-list-objects-in-bucket-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "s3:ListBucket"
+                Resource:
+                  - !Sub "arn:aws:s3:::biomage-source-${Environment}"
+
```

Comment:
This is added directly from `irsa-ui-role.yaml` to this CF file instead of having a separate one. The role is linked together with Cognito and it is not similar to other IRSA roles (which specify what service accounts in kubernetes can run on the cluster).